### PR TITLE
Add a new check for empty projects in projects.xml

### PR DESCRIPTION
**Pull request by mgorny**
@radhermit, I can't figure out how to make the test actually run.

### CODE REVIEW

File: tests/data/standalone/ProjectMetadataCheck/EmptyProject/expected [hunk 1]
```diff
@@ -0,0 +1,2 @@
+repo
+  EmptyProject: 'empty@example.com' project has no members
```
Comment (mgorny) [inline tests/data/standalone/ProjectMetadataCheck/EmptyProject/expected]: And yes, I know this is wrong output. I'd liek to see it fail first, though.

**Comment by mgorny** [comment]
Heh, `git clean -dfx` helped. It's ready for review, @radhermit.

File: src/pkgcheck/checks/repo_metadata.py [hunk 2]
```diff
@@ -688,3 +688,35 @@ def feed(self, pkgs):
...
+    ])
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.repo = self.options.target_repo
+
+    def finish(self):
+        for key, project in self.repo.projects_xml.projects.items():
+            if not project.recursive_members:
+                # TODO: make it serializable
```
Comment (radhermit) [inline src/pkgcheck/checks/repo_metadata.py]: What's this comment in regards to? Making project object instances serializable?

Comment (mgorny) [reply src/pkgcheck/checks/repo_metadata.py]: Hmm, I can't reproduce now. I'm gonna repush and see what happens.

Comment (mgorny) [reply src/pkgcheck/checks/repo_metadata.py]: You can see the error in travis now. Lemme know if I should fix the Project instance somehow or you want to fix it on `to_json` end instead.

**Comment by radhermit** [comment]
Thanks, merged in https://github.com/pkgcore/pkgcheck/commit/04f4a5abffc2136c91718d7780cb75fee9a2ce57.

**Pull request closed without merging**